### PR TITLE
feat: add `parse transactions` cmd 

### PR DESCRIPTION
**Pull request by MonikaCat**
## Checklist
- [x] Targeted PR against correct branch.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit tests.  
- [x] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: database/postgresql/postgresql.go [hunk 1]
```diff
@@ -130,7 +130,18 @@ func (db *Database) saveTxInsidePartition(tx *types.Tx, partitionId int64) error
 	sqlStatement := `
 INSERT INTO transaction 
 (hash, height, success, messages, memo, signatures, signer_infos, fee, gas_wanted, gas_used, raw_log, logs, partition_id) 
-VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13) ON CONFLICT DO NOTHING`
+VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13) ON CONFLICT (hash) DO UPDATE
```
Comment (huichiaotsou) [inline database/postgresql/postgresql.go]:
```suggestion
VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13) ON CONFLICT (hash, partition_id) DO UPDATE 
```

File: database/postgresql/postgresql.go [hunk 2]
```diff
@@ -248,7 +259,12 @@ func (db *Database) SaveMessage(msg *types.Message) error {
 func (db *Database) saveMessageInsidePartition(msg *types.Message, partitionID int64) error {
 	stmt := `
 INSERT INTO message(transaction_hash, index, type, value, involved_accounts_addresses, height, partition_id) 
-VALUES ($1, $2, $3, $4, $5, $6, $7) ON CONFLICT DO NOTHING`
+VALUES ($1, $2, $3, $4, $5, $6, $7) ON CONFLICT (transaction_hash) DO UPDATE
```
Comment (huichiaotsou) [inline database/postgresql/postgresql.go]:
```suggestion
VALUES ($1, $2, $3, $4, $5, $6, $7) ON CONFLICT (transaction_hash, index, partition_id) DO UPDATE 
```

File: database/postgresql/postgresql.go [hunk 3]
```diff
@@ -248,7 +259,12 @@ func (db *Database) SaveMessage(msg *types.Message) error {
...
+		SET height = excluded.height, 
+			type = excluded.type,
+			value = excluded.value,
+			involved_accounts_addresses = excluded.involved_accounts_addresses,
+			height = excluded.height`
```
Comment (huichiaotsou) [inline database/postgresql/postgresql.go]:
```suggestion
		SET height = excluded.height, 
			type = excluded.type,
			value = excluded.value,
			involved_accounts_addresses = excluded.involved_accounts_addresses,
```

Comment (huichiaotsou) [reply database/postgresql/postgresql.go]: height is set twice :D

File: cmd/parse/transactions/transactions.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+			if end > 0 {
+				endHeight = end
+			}
+
+			log.Info().Int64("start height", startHeight).Int64("end height", endHeight).
+				Msg("getting transactions...")
+			for k := startHeight; k <= endHeight; k++ {
+				err = worker.ProcessTransactions(k)
+				if err != nil {
+					return fmt.Errorf("error while re-fetching transactions %d: %s", k, err)
```
Comment (huichiaotsou) [inline cmd/parse/transactions/transactions.go]:
```suggestion
					return fmt.Errorf("error while re-fetching transactions of height %d: %s", k, err)
```

File: cmd/parse/transactions/transactions.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+				return fmt.Errorf("error while getting chain latest block height: %s", err)
+			}
+			if end > 0 {
+				endHeight = end
+			}
+
+			log.Info().Int64("start height", startHeight).Int64("end height", endHeight).
+				Msg("getting transactions...")
+			for k := startHeight; k <= endHeight; k++ {
+				err = worker.ProcessTransactions(k)
```
Comment (huichiaotsou) [inline cmd/parse/transactions/transactions.go]:
```suggestion
			for k := startHeight; k <= endHeight; k++ {
				log.Info().Int64("height", k).Msg("processing transactions...")
				err = worker.ProcessTransactions(k)
```

**Review by huichiaotsou — APPROVED** [review]
tested ACK

**Review by rmontagnin — APPROVED** [review]

**Pull request merged**